go/vt/vtgate/planbuilder/dml.go

```diff
@@ -184,6 +243,18 @@ func getDMLRouting(where *sqlparser.Where, route *engine.Route) error {
 	return errors.New("unsupported: multi-shard where clause in DML")
 }
 
+func extractValueFromUpdate(upd *sqlparser.UpdateExpr, col sqlparser.ColIdent) (pv sqltypes.PlanValue, ok bool) {
```

Comment:
Please add a comment explaining what the intent of this function is and what happens on the various error cases.

Reply:
It may actually be better to make this function return an error.